llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -11739,8 +11750,9 @@ InstructionCost BoUpSLP::getTreeCost(ArrayRef<Value *> VectorizedVals) {
       ExtraCost = TTI->getExtractWithExtendCost(Extend, EU.Scalar->getType(),
                                                 VecTy, EU.Lane);
     } else {
-      ExtraCost = TTI->getVectorInstrCost(Instruction::ExtractElement, VecTy,
-                                          CostKind, EU.Lane);
+      ExtraCost = TTI->getVectorInstrCost(
+          Instruction::ExtractElement, VecTy, CostKind, EU.Lane, nullptr,
```

Comment:
To be honest I'm a bit confused by the interface here. It seems like the only use case for this function is when the Op0 and Op1 arguments are both nullptr. So why not just remove these two arguments from the interface?

Reply:
This function is called from multiple places and at some places, Op0 and Op1 are indeed not null.

Reply:
Oh ok, I looked at all calls to getVectorInstrCost outside of the TargetTransformInfo files (which simply pass through the operands in a chain) and it wasn't obvious to me where we passed in non-null. I can see examples like this:

```
InstructionCost TargetTransformInfo::getVectorInstrCost
```

but these pass the same operands from one interface to another.

Reply:
SLPVectorizer has other calls to this API where Op0 and Op1 are not null

Reply:
ah yes, got confused. My bad. will remove Op0 and Op1